### PR TITLE
Disable persistence by default (unless --enable-persistence passed)

### PR DESCRIPTION
**Pull request by geelen**
Thought it was time to pull my finger out and send a PR, now that I don't have the "but I don't know Rust" defence to lean back on...

### CODE REVIEW

**Comment by geelen** [comment]
This resolves #320

**Comment by threepointone** [comment]
call it `--experimental-enable-local-persistence`? So we can unblock you, the flag is clearer, and we can iterate on this if needed. I'm coming around to the idea that we should disable persistence for `dev`.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "printWidth": 80,
+  "singleQuote": false,
+  "semi": true
+}
```
Comment (petebacondarwin) [inline .prettierrc:5]: Perhaps this change is something to discuss in a separate PR?

File: packages/wrangler/src/dev.tsx [hunk 2]
```diff
@@ -726,43 +731,43 @@ type useHotkeysInitialState = {
...
+  //       case "q": // shut down
+  //       case "x": // shut down
+  //         process.exit(0);
+  //         break;
+  //       default:
+  //         // nothing?
+  //         break;
+  //     }
+  //   }
+  // );
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev.tsx:770]: Is this supposed to be commented out?

File: .prettierrc (hunk 1 above)
Comment (petebacondarwin) [reply .prettierrc:5]: Oh I see, this is to match the current settings, and override any user-level config that a developer might have...?

File: packages/wrangler/src/dev.tsx [hunk 3]
```diff
@@ -40,6 +40,7 @@ export type DevProps = {
   initialMode: "local" | "remote";
   jsxFactory: undefined | string;
   jsxFragment: undefined | string;
+  enablePersistence: boolean;
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev.tsx:43]: BIKESHED: `enableLocalPersistence` or `localModePersistence` or something that signifies that this only affects local mode?

**Review by petebacondarwin — COMMENTED** [review]
We'll be needing a change-set too!
Run the following in the root of the project.

```bash
npx changeset
```

Choosing to bump the wrangler package by only a patch version (since we are in 0.x version mode)

File: .prettierrc (hunk 1 above)
Comment (geelen) [reply .prettierrc:5]: Yeah my `~/.prettierrc` is set to the opposite of all of these, so running prettier on anything exploded the diffs. I ran `npm run prettify` after adding this and it changed nothing, except the `npm run check:format` on CI still failed. Not sure why...

File: packages/wrangler/src/dev.tsx (hunk 2 above)
Comment (geelen) [reply packages/wrangler/src/dev.tsx:770]: Damn sorry that was me trying to fix something else... Will redo and repush

**Review by petebacondarwin — APPROVED** [review]
LGTM - the failure is a TypeScript one...

```
Error: [check:type  ] packages/wrangler/src/__tests__/dev.test.tsx(49,6): error TS2741: Property 'enableLocalPersistence' is missing in type '{ name: string | undefined; entry: string; port: number | undefined; buildCommand: { command?: string | undefined; cwd?: string | undefined; watch_dir?: string | undefined; }; format: CfScriptFormat; ... 9 more ...; bindings: { ...; }; }' but required in type 'DevProps'.
```

Looks like you need to update `dev.test.ts`

File: .changeset/two-coins-compare.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Disabled local persistence by default & added --experimental-enable-local-persistence flag
```
Comment (petebacondarwin) [inline .changeset/two-coins-compare.md:5]: Oh last thing... Could you add a breaking changes notification here... Something like:

```
BREAKING CHANGE:

When running `dev` locally any data stored in KV, Durable Objects or the cache are no longer persisted between sessions by default.

To turn this back on add the `--experimental-enable-local-persistence` at the command line.
```

Comment (geelen) [reply .changeset/two-coins-compare.md]: Like this?

**Review by petebacondarwin — APPROVED** [review]
:shipit:

**Comment by petebacondarwin** [comment]
@geelen - oops:

```
[check:lint  ] /Users/pbacondarwin/dev/wrangler2/packages/wrangler/src/dev.tsx
[check:lint  ]   320:6  warning  React Hook useEffect has a missing dependency: 'props.enableLocalPersistence'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

**Comment by threepointone** [comment]
fix for the lint warning included in https://github.com/cloudflare/wrangler2/pull/333

**Pull request merged**